agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/diagnostics/InstrumentationKeyFinder.java

```diff
@@ -0,0 +1,41 @@
+package com.microsoft.applicationinsights.agentc.internal.diagnostics;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.microsoft.applicationinsights.agentc.internal.model.Global;
+import com.microsoft.applicationinsights.internal.config.TelemetryConfigurationFactory;
+import com.microsoft.applicationinsights.internal.config.connection.ConnectionString.Keywords;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InstrumentationKeyFinder implements DiagnosticsValueFinder {
+    @Nonnull
+    @Override
+    public String getName() {
+        return "ikey";
+    }
+
+    @Nullable
+    @Override
+    public String getValue() {
+        try {
+            return Global.getTelemetryClient().getContext().getInstrumentationKey();
+        } catch (NullPointerException npe) {
```

Comment:
catching `NullPointerException` looks odd, can we null check somehow instead?

also, are we checking this just for performance vs just always using getting from env vars below?

Reply:
`Global.getTelemetryClient()` throws npe if client is not set. getContext is never null.

I was thinking this would be the most reliable, but we can defer to env vars if you think that is sufficient.